Traceback (most recent call last):
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/nbclient/client.py", line 1117, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/nbclient/util.py", line 78, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/nbclient/util.py", line 57, in just_run
    return loop.run_until_complete(coro)
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/nbclient/client.py", line 558, in async_execute
    await self.async_execute_cell(
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/nbclient/client.py", line 862, in async_execute_cell
    self._check_raise_for_error(cell, exec_reply)
  File "/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/nbclient/client.py", line 765, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# load testing file
test_files = ['root://eospublic.cern.ch//eos/opendata/cms/datascience/HiggsToBBNtupleProducerTool/HiggsToBBNTuple_HiggsToBB_QCD_RunII_13TeV_MC/test/ntuple_merged_0.root']
test_generator = DataGenerator(test_files, features, labels, spectators, batch_size=1024, n_dim=ntracks, 
                               remove_mass_pt_window=True, 
                               remove_unlabeled=True)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mOSError[0m                                   Traceback (most recent call last)
[0;32m/tmp/ipykernel_4036/67902189.py[0m in [0;36m<module>[0;34m[0m
[1;32m      1[0m [0;31m# load testing file[0m[0;34m[0m[0;34m[0m[0m
[1;32m      2[0m [0mtest_files[0m [0;34m=[0m [0;34m[[0m[0;34m'root://eospublic.cern.ch//eos/opendata/cms/datascience/HiggsToBBNtupleProducerTool/HiggsToBBNTuple_HiggsToBB_QCD_RunII_13TeV_MC/test/ntuple_merged_0.root'[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 3[0;31m test_generator = DataGenerator(test_files, features, labels, spectators, batch_size=1024, n_dim=ntracks, 
[0m[1;32m      4[0m                                [0mremove_mass_pt_window[0m[0;34m=[0m[0;32mTrue[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m      5[0m                                remove_unlabeled=True)

[0;32m~/work/capstone-particle-physics-domain/capstone-particle-physics-domain/weeks/DataGenerator.py[0m in [0;36m__init__[0;34m(self, list_files, features, labels, spectators, batch_size, n_dim, remove_mass_pt_window, remove_unlabeled, return_spectators, max_entry, scale_mass_pt)[0m
[1;32m     30[0m         [0mrunning_total[0m [0;34m=[0m [0;36m0[0m[0;34m[0m[0;34m[0m[0m
[1;32m     31[0m         [0;32mfor[0m [0mi[0m[0;34m,[0m [0mfile_name[0m [0;32min[0m [0menumerate[0m[0;34m([0m[0mself[0m[0;34m.[0m[0mlist_files[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 32[0;31m             [0mroot_file[0m [0;34m=[0m [0muproot[0m[0;34m.[0m[0mopen[0m[0;34m([0m[0mfile_name[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     33[0m             [0mself[0m[0;34m.[0m[0mopen_files[0m[0;34m.[0m[0mappend[0m[0;34m([0m[0mroot_file[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     34[0m             [0mtree[0m [0;34m=[0m [0mroot_file[0m[0;34m[[0m[0;34m'deepntuplizer/tree'[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m

[0;32m/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/uproot/reading.py[0m in [0;36mopen[0;34m(path, object_cache, array_cache, custom_classes, decompression_executor, interpretation_executor, **options)[0m
[1;32m    147[0m         )
[1;32m    148[0m [0;34m[0m[0m
[0;32m--> 149[0;31m     file = ReadOnlyFile(
[0m[1;32m    150[0m         [0mfile_path[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m    151[0m         [0mobject_cache[0m[0;34m=[0m[0mobject_cache[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m

[0;32m/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/uproot/reading.py[0m in [0;36m__init__[0;34m(self, file_path, object_cache, array_cache, custom_classes, decompression_executor, interpretation_executor, **options)[0m
[1;32m    587[0m             [0mfile_path[0m[0;34m,[0m [0mself[0m[0;34m.[0m[0m_options[0m[0;34m[0m[0;34m[0m[0m
[1;32m    588[0m         )
[0;32m--> 589[0;31m         self._source = Source(
[0m[1;32m    590[0m             [0mfile_path[0m[0;34m,[0m [0;34m**[0m[0mself[0m[0;34m.[0m[0m_options[0m  [0;31m# NOTE: a comma after **options breaks Python 2[0m[0;34m[0m[0;34m[0m[0m
[1;32m    591[0m         )

[0;32m/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/uproot/source/xrootd.py[0m in [0;36m__init__[0;34m(self, file_path, **options)[0m
[1;32m    264[0m         [0mself[0m[0;34m.[0m[0m_file_path[0m [0;34m=[0m [0mfile_path[0m[0;34m[0m[0;34m[0m[0m
[1;32m    265[0m         [0mself[0m[0;34m.[0m[0m_num_bytes[0m [0;34m=[0m [0;32mNone[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 266[0;31m         [0mself[0m[0;34m.[0m[0m_open[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    267[0m [0;34m[0m[0m
[1;32m    268[0m     [0;32mdef[0m [0m_open[0m[0;34m([0m[0mself[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;32m/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/uproot/source/xrootd.py[0m in [0;36m_open[0;34m(self)[0m
[1;32m    267[0m [0;34m[0m[0m
[1;32m    268[0m     [0;32mdef[0m [0m_open[0m[0;34m([0m[0mself[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 269[0;31m         [0mself[0m[0;34m.[0m[0m_resource[0m [0;34m=[0m [0mXRootDResource[0m[0;34m([0m[0mself[0m[0;34m.[0m[0m_file_path[0m[0;34m,[0m [0mself[0m[0;34m.[0m[0m_timeout[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    270[0m [0;34m[0m[0m
[1;32m    271[0m         [0;31m# this ThreadPool does not need a resource, it's only used to submit[0m[0;34m[0m[0;34m[0m[0m

[0;32m/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/uproot/source/xrootd.py[0m in [0;36m__init__[0;34m(self, file_path, timeout)[0m
[1;32m     80[0m         [0mself[0m[0;34m.[0m[0m_file_path[0m [0;34m=[0m [0mfile_path[0m[0;34m[0m[0;34m[0m[0m
[1;32m     81[0m         [0mself[0m[0;34m.[0m[0m_timeout[0m [0;34m=[0m [0mtimeout[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 82[0;31m         [0mself[0m[0;34m.[0m[0m_open[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     83[0m [0;34m[0m[0m
[1;32m     84[0m     [0;32mdef[0m [0m_open[0m[0;34m([0m[0mself[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;32m/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/uproot/source/xrootd.py[0m in [0;36m_open[0;34m(self)[0m
[1;32m     89[0m         [0mstatus[0m[0;34m,[0m [0mdummy[0m [0;34m=[0m [0mself[0m[0;34m.[0m[0m_file[0m[0;34m.[0m[0mopen[0m[0;34m([0m[0mself[0m[0;34m.[0m[0m_file_path[0m[0;34m,[0m [0mtimeout[0m[0;34m=[0m[0mself[0m[0;34m.[0m[0m_xrd_timeout[0m[0;34m([0m[0;34m)[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     90[0m         [0;32mif[0m [0mstatus[0m[0;34m.[0m[0merror[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 91[0;31m             [0mself[0m[0;34m.[0m[0m_xrd_error[0m[0;34m([0m[0mstatus[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     92[0m [0;34m[0m[0m
[1;32m     93[0m     [0;32mdef[0m [0m__getstate__[0m[0;34m([0m[0mself[0m[0;34m)[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m

[0;32m/usr/share/miniconda/envs/ml-latest/lib/python3.9/site-packages/uproot/source/xrootd.py[0m in [0;36m_xrd_error[0;34m(self, status)[0m
[1;32m    115[0m [0;34m[0m[0m
[1;32m    116[0m         [0;32melse[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 117[0;31m             raise OSError(
[0m[1;32m    118[0m                 """XRootD error: {0}
[1;32m    119[0m in file {1}""".format(

[0;31mOSError[0m: XRootD error: [ERROR] Operation expired
in file root://eospublic.cern.ch//eos/opendata/cms/datascience/HiggsToBBNtupleProducerTool/HiggsToBBNTuple_HiggsToBB_QCD_RunII_13TeV_MC/test/ntuple_merged_0.root
OSError: XRootD error: [ERROR] Operation expired
in file root://eospublic.cern.ch//eos/opendata/cms/datascience/HiggsToBBNtupleProducerTool/HiggsToBBNTuple_HiggsToBB_QCD_RunII_13TeV_MC/test/ntuple_merged_0.root

